crates/key-value-azure/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use azure_data_cosmos::{
+    prelude::{AuthorizationToken, CollectionClient, CosmosClient, Query},
+    CosmosEntity,
+};
+use futures::StreamExt;
+use serde::{Deserialize, Serialize};
+use spin_core::async_trait;
+use spin_key_value::{log_error, Error, Store, StoreManager};
+use tokio::{
+    sync::{Mutex, OnceCell},
+    task,
+};
+
+pub struct KeyValueAzureCosmos {
+    key: String,
+    account: String,
+    database: String,
+    collection: String,
+    pub client: OnceCell<Arc<Mutex<CollectionClient>>>,
+}
+
+impl KeyValueAzureCosmos {
+    pub fn new(key: String, account: String, database: String, collection: String) -> Result<Self> {
+        Ok(Self {
+            key,
+            account,
+            database,
+            collection,
+            client: OnceCell::new(),
+        })
+    }
+}
+
+#[async_trait]
+impl StoreManager for KeyValueAzureCosmos {
+    async fn get(&self, name: &str) -> Result<Arc<dyn Store>, Error> {
+        let client = task::block_in_place(|| {
```

Comment:
Sorry if this is a stupid question, but why is it using a blocking call in an async func?

Reply:
Ooh, good catch. That was a left-over from an early implementation of this.
Fixed.